### PR TITLE
V24.2.0 alpha.2 release notes

### PR DESCRIPTION
**Pull request by florence-crl**
REL-1081

edited notes

### CODE REVIEW

File: src/current/_includes/releases/v24.2/v24.2.0-alpha.2.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  - [`storage.sstable.compression_algorithm_backup_storage`]({% link v24.2/cluster-settings.md %}#setting-storage-sstable-compression-algorithm-backup-storage)
+  - [`storage.sstable.compression_algorithm_backup_transport`]({% link v24.2/cluster-settings.md %}#setting-storage-sstable-compression-algorithm-backup-transport) [#126508][#126508]
+
+<h3 id="v24-2-0-alpha-2-sql-language-changes">SQL language changes</h3>
+
+- Added the [`sql.auth.grant_option_inheritance.enabled` cluster setting]({% link v24.2/cluster-settings.md %}#setting-sql-auth-grant-option-inheritance-enabled). The default value is `true`, which maintains consistency with CockroachDB's previous behavior - users granted a privilege with [`WITH GRANT OPTION`]({% link v24.2/grant.md %}) can in turn grant that privilege to others. When `sql.auth.grant_option_inheritance.enabled` is set to `false`, the `GRANT OPTION` is not inherited through role membership, thereby preventing descendant roles from granting the privilege to others. However, the privilege itself continues to be inherited through role membership. [#125590][#125590]
```
Comment (kathancox) [inline src/current/_includes/releases/v24.2/v24.2.0-alpha.2.md:16]:
```suggestion
- Added the [`sql.auth.grant_option_inheritance.enabled` cluster setting]({% link v24.2/cluster-settings.md %}#setting-sql-auth-grant-option-inheritance-enabled). The default value is `true`, which maintains consistency with CockroachDB's previous behavior: users granted a privilege with [`WITH GRANT OPTION`]({% link v24.2/grant.md %}) can in turn grant that privilege to others. When `sql.auth.grant_option_inheritance.enabled` is set to `false`, the `GRANT OPTION` is not inherited through role membership, thereby preventing descendant roles from granting the privilege to others. However, the privilege itself continues to be inherited through role membership. [#125590][#125590]
```

File: src/current/_includes/releases/v24.2/v24.2.0-alpha.2.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+<h3 id="v24-2-0-alpha-2-bug-fixes">Bug fixes</h3>
+
+- Fixed a bug that could cause internal errors when a [routine]({% link v24.2/user-defined-functions.md %}) had polymorphic parameters or a polymorphic return type or both. The bug has existed since v22.2 when [user-defined functions (UDFs)]({% link v24.2/user-defined-functions.md %}) were introduced. [#123459][#123459]
+- In [`SHOW CREATE`]({% link v24.2/show-create.md %}) output, the name of an [enum type]({% link v24.2/enum.md %}) is now formatted as a two-part name (`schema.enum_type`) instead of a three-part name (`database.schema.enum_type`). This change makes it easier to apply the output with enum types to other databases. [#125996][#125996]
+- When [altering the data type of columns]({% link v24.2/alter-table.md %}#alter-column) with the [hidden attribute (`NOT VISIBLE`)]({% link v24.2/create-table.md %}#not-visible-property), the alteration now preserves the hidden attribute in the column. Additionally, type alterations for columns with [ON UPDATE]({% link v24.2/create-table.md %}#on-update-expressions) expressions or [DEFAULT]({% link v24.2/default-value.md %}) expressions are now allowed. [#126033][#126033]
```
Comment (kathancox) [inline src/current/_includes/releases/v24.2/v24.2.0-alpha.2.md:23]:
```suggestion
- When [altering the data type of columns]({% link v24.2/alter-table.md %}#alter-column) with the [hidden attribute (`NOT VISIBLE`)]({% link v24.2/create-table.md %}#not-visible-property), the alteration now preserves the hidden attribute in the column. Additionally, type alterations for columns with [`ON UPDATE`]({% link v24.2/create-table.md %}#on-update-expressions) expressions or [`DEFAULT`]({% link v24.2/default-value.md %}) expressions are now allowed. [#126033][#126033]
```

**Review by kathancox — APPROVED** [review]
Couple of nits, but LGTM!

File: src/current/_includes/releases/v24.2/v24.2.0-alpha.2.md (hunk 1 above)
Comment (florence-crl) [reply src/current/_includes/releases/v24.2/v24.2.0-alpha.2.md]: updated

File: src/current/_includes/releases/v24.2/v24.2.0-alpha.2.md (hunk 2 above)
Comment (florence-crl) [reply src/current/_includes/releases/v24.2/v24.2.0-alpha.2.md]: updated

**Review by florence-crl — COMMENTED** [review]
TFTR!

**Pull request merged**